File: nc-scripts/src/app/dto/workflows/workflowdetails.ts
import { Workflow } from './workflow';
import { WorkflowNode } from './workflownode';
import { Transition } from './transition';

/**
 * full workflow data with all nodes and transitions
 */
export interface WorkflowDetails extends Workflow {

    /**
     * nodes in workflow
     */
    nodes: WorkflowNode[],

    /**
     * transitions in workflow
     */
    transitions: Transition[]
}

File: ScriptService/Dto/Workflows/WorkflowDetails.cs
namespace ScriptService.Dto.Workflows {

    /// <summary>
    /// detailed workflow information
    /// </summary>
    public class WorkflowDetails : Workflow {

        /// <summary>
        /// nodes contained in workflow
        /// </summary>
        public NodeDetails[] Nodes { get; set; }

        /// <summary>
        /// transitions of nodes contained in workflow
        /// </summary>
        public TransitionData[] Transitions { get; set; }
    }
}